### PR TITLE
refactor: Update fetching instructions in test PR guide

### LINKED ISSUE #1072: Update steps in how to test a PR doc
## Description
As a developer,
I need the quality assurance doc of how to test a PR with correct steps,
so that it'll help everyone in testing PRs locally.
We shouldn't be cloning and checking out the author's fork we can directly use

` git fetch origin pull/ID/head:BRANCHNAME`

Read [this reference doc](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) for more info

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
10mins

### PR DESCRIPTION
**Pull request by keshavgbpecdelhi**
### Description

We shouldn't be cloning and checking out the author's fork we can directly use
```
git fetch origin pull/ID/head:BRANCHNAME

```

Fixes #1072

### Type of Change:

- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

It's being guessed and assumed, so no tests.

- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by keshavgbpecdelhi** [comment]
Did mistake?

File: docs/test-pr-guide.md
Comment (devkapilbansal) [inline docs/test-pr-guide.md]:
```suggestion
git fetch upstream pull/ID/head:<branch_name>
```

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Thanks for this @keshavgbpecdelhi 
Please make the required changes

File: docs/test-pr-guide.md
Comment (devkapilbansal) [inline docs/test-pr-guide.md]: This line is also not necessary.
```suggestion
```

File: docs/test-pr-guide.md
Comment (devkapilbansal) [inline docs/test-pr-guide.md:16]: Would be better if we can explain what ID stands for here :thinking:

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Just some minor improvements :sweat_smile:

File: docs/test-pr-guide.md
Comment (keshavgbpecdelhi) [reply docs/test-pr-guide.md:16]: Sir how to do this, 🤔

Comment (devkapilbansal) [reply docs/test-pr-guide.md:16]: First of all, no need to use sir.

File: docs/test-pr-guide.md
Comment (devkapilbansal) [inline docs/test-pr-guide.md]: Although it's debatable, we can change it to something like
```suggestion
5. Run the following commands to get to the PR branch, where `<id>` is the PR id and `<branch_name>` is the random name for the branch.
```

CC:- @anitab-org/mentorship-maintainers

Comment (epicadk) [reply docs/test-pr-guide.md]: I just use github cli so can't really help with this.

Comment (devkapilbansal) [reply docs/test-pr-guide.md]: @keshavgbpecdelhi please make the required changes

File: docs/test-pr-guide.md
Comment (epicadk) [inline docs/test-pr-guide.md:14]: Also add "Alternatively if you use the github cli you can run the following command `gh pr checkout <pr_ number>`"

**Review by devkapilbansal — APPROVED** [review]
LGTM :+1: 
Thanks @keshavgbpecdelhi for this

I tested the command locally and it worked. With this we don't need new contributors to change the name of branch if they accidentally use default branch. Previously, it was causing problems in testing but now, we can decide name of branch while fetching

File: docs/test-pr-guide.md
Comment (devkapilbansal) [reply docs/test-pr-guide.md:14]: Do we need to use github cli now?  It doesn't allow changing the branch name at now

Comment (epicadk) [reply docs/test-pr-guide.md:14]: We don't need to it's just an alternative to this. I don't see why we'd need to change the branch name while reviewing a pr 😕.

Comment (devkapilbansal) [reply docs/test-pr-guide.md:14]: Most of the times, new Contributor made PR from branch named `develop` and while trying to fetch locally it cause errors. Anyways, it's my personal opinion that we should change branch name .

Comment (epicadk) [reply docs/test-pr-guide.md:14]: > Most of the times, new Contributor made PR from branch named `develop` and while trying to fetch locally it cause errors. Anyways, it's my personal opinion that we should change branch name . 

Sorry I still don't get it. The branch name would be whatever the contributor chooses right? I don't see how it affects the testing.

**Review by vj-codes — APPROVED** [review]
Congrats on your first contribution to the repo @keshavgbpecdelhi 🎉 
LGTM!

**Comment by vj-codes** [comment]
@isabelcosta can you approve the workflow run to cross-check although no code has been changed , once passed this can be merged

**Comment by devkapilbansal** [comment]
Adding `Ready to Merge` as code is unchanged

**Comment by devkapilbansal** [comment]
@keshavgbpecdelhi please change the PR title to make it more descriptive.
Eg:- `refactor: Update fetching instructions in test PR guide`

**Comment by keshavgbpecdelhi** [comment]
LGTYN?

**Review by isabelcosta — APPROVED** [review]
This is amazing! I just learned something new here! Game changer 🤩 
Thank you @keshavgbpecdelhi for working on this 🙌🏾

**Comment by epicadk** [comment]
I'd actually asked for a small change :/ . No worries will open another issue.

**Comment by isabelcosta** [comment]
@epicadk yes yes please do create an issue! I saw the comment briefly but did not notice you actually asked for a change.

**Pull request merged**
